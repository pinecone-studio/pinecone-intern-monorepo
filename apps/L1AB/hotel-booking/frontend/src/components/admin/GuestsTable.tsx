'use client';
import React from 'react';
import { Select, SelectContent, SelectItem, SelectTrigger, SelectValue } from '@/components/ui/select';
import { ChevronDown, ChevronLeft, ChevronRight, ChevronsLeft, ChevronsRight } from 'lucide-react';
import { Button } from '@/components/ui/button';
import { Table, TableBody, TableCell, TableHead, TableHeader, TableRow } from '@/components/ui/table';

const invoices = [
  {
    ID: '001',
    Name: 'Baatar Erdenebat',
    Hotel: 'Chingis Khan Hotel',
    Rooms: 'Economy Double Room, City View',
    Guests: '1 Adult',
    Date: 'Nov 5 - Nov 7',
    Status: 'Booked',
  },
  {
    ID: '002',
    Name: 'Tsetsenbayar Munkhbat',
    Hotel: 'Chingis Khan Hotel',
    Rooms: 'Standard Twin Room, City View',
    Guests: '2 Adult',
    Date: 'Jan 10 - Jan 12',
    Status: 'Completed',
  },
  {
    ID: '003',
    Name: 'Munkhbayar Tserendorj',
    Hotel: 'Chingis Khan Hotel',
    Rooms: 'Deluxe Twin Room, City View',
    Guests: '2 Adult',
    Date: 'Jul 4 - Jul 6',

    Status: 'Cancelled',
  },
  {
    ID: '004',
    Name: 'Sarnai Erdenetsetsegd',
    Hotel: 'Chingis Khan Hotel',
    Rooms: 'Economy Double Room, City View',
    Guests: '1 Adult',
    Date: 'Dec 5 - Dec 7',
    Status: 'Cancelled',
  },
  {
    ID: '005',
    Name: 'Nomin Erdenebold',
    Hotel: 'Chingis Khan Hotel',
    Rooms: 'Deluxe Twin Room, City View',
    Guests: '2 Adult',
    Date: 'May 30 - Jun 1',
    Status: 'Completed',
  },
];
export const GuestsTable = () => {
  return (
    <div className=" p-3 flex flex-col">
      <div className="my-4">Guests</div>
      <div className="flex gap-4 my-4">
        <div className="flex w-full">
          <Button variant="outline" className="border border-[#d6d6df] text-[#71717A] w-full justify-start">
            Search
          </Button>
        </div>
        <div className="flex">
          <Select>
            <SelectTrigger className="w-[180px]">
              <SelectValue placeholder="Status" />
            </SelectTrigger>
            <SelectContent>
              <div className="flex gap-2 justify-center items-center text-[#71717A] p-1">
                <ChevronDown />
                Search option...
              </div>
              <div className=" flex px-4 gap-3">
                <div className="h-full w-full border border-[#9797aa]"></div>
              </div>
              <SelectItem value="all">All</SelectItem>
              <SelectItem value="Booked">Booked</SelectItem>
              <SelectItem value="Cancelled">Cancelled</SelectItem>
              <SelectItem value="Visited">Visited</SelectItem>
            </SelectContent>
          </Select>
        </div>
      </div>
      <div>
        <Table>
          <TableHeader>
            <TableRow>
              <TableHead className="w-[100px]">ID</TableHead>
              <TableHead>Name</TableHead>
              <TableHead>Hotel</TableHead>
              <TableHead>Rooms</TableHead>
              <TableHead>Guests</TableHead>
              <TableHead>Date</TableHead>
              <TableHead>Status</TableHead>
            </TableRow>
          </TableHeader>
          <TableBody>
            {invoices.map((invoice) => (
              <TableRow key={invoice.ID}>
                <TableCell className="font-medium">{invoice.ID}</TableCell>
                <TableCell>{invoice.Name}</TableCell>
                <TableCell>{invoice.Hotel}</TableCell>
                <TableCell>{invoice.Rooms}</TableCell>
                <TableCell>{invoice.Guests}</TableCell>
                <TableCell>{invoice.Date}</TableCell>
                <TableCell className="">
                  <button className="bg-[#2563EB] p-2 rounded-xl text-[#FAFAFA]">{invoice.Status}</button>
                </TableCell>
              </TableRow>
            ))}
          </TableBody>
        </Table>
        <div className="flex justify-between p-6 ">
          <div className="text-[#F4F4F5]">.</div>
          <div className="flex flex-row gap-4">
            <div>
              <h1> Page 1 of 10</h1>
            </div>
            <div className="flex gap-2 ">
              <button className="w-[32px] h-[32px] border flex justify-center items-center rounded-md">
                <ChevronsLeft />
              </button>
              <button className="w-[32px] h-[32px] border flex justify-center items-center rounded-md">
                <ChevronLeft />
              </button>
              <button className="w-[32px] h-[32px] border flex justify-center items-center rounded-md">
                <ChevronRight />
              </button>
              <button className="w-[32px] h-[32px] border flex justify-center items-center rounded-md">
                <ChevronsRight />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};
