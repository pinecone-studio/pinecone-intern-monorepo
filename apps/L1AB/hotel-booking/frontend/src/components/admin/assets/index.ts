export * from './SideBar';
export * from './Header';
export * from './Footer';
export * from './HotelsSearchBar';
