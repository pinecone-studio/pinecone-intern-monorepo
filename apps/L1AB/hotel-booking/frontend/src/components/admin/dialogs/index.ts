export * from './LocationDialog';
export* from './GeneralInfoDialog';
export * from './HotelGeneralInfoDialog';
export * from './HotelNameDialog';
export * from './RoomServicesDialog';
