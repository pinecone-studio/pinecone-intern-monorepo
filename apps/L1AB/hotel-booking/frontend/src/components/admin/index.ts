export * from './Container';
export * from './HotelsTable';
export * from './GuestsTable';
export * from './HotelDetails';
export * from './RoomDetails';
export * from './GuestInfo';
export * from './UploadPhoto';
