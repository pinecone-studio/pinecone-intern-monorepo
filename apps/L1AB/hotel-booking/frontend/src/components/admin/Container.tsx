import React, { PropsWithChildren } from 'react'
import { Footer, Header, SideBar } from './assets'

export const Container = ({ children }: PropsWithChildren) => {
    return (
        <div className="w-screen h-screen flex fixed">
            <SideBar />
            <div className="bg-secondary w-full flex flex-col justify-between">
                <Header />
                <div style={{ height: 'calc(100vh - 112px)' }} className='overflow-hidden'>
                    {children}
                </div>
                <Footer />
            </div>
        </div>
    )
}
