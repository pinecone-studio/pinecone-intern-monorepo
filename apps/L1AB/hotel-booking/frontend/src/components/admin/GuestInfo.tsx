import React from 'react';
import { DetailsCard, DetailsContainer, DetailsLeft, DetailsRight } from './assets';

export const GuestInfo = () => {
  return (
    <DetailsContainer name={'Shagai Nyamdorj'}>
      <DetailsLeft>
        <DetailsCard>
          <div>Guest Info</div>
        </DetailsCard>
      </DetailsLeft>
      <DetailsRight>
        <DetailsCard>
          <div>Economy Single Room</div>
        </DetailsCard>
        <DetailsCard>
          <div>Price Detail</div>
        </DetailsCard>
      </DetailsRight>
    </DetailsContainer>
  );
};
