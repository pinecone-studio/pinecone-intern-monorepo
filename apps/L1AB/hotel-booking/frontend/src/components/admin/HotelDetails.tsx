'use client';

import React from 'react';
import { DetailsCard, DetailsContainer, DetailsLeft, DetailsRight } from './assets';
import {
  HotelDetailsAboutThisProperty,
  HotelDetailsAmenities,
  HotelDetailsFrequently,
  HotelDetailsGeneralInfo,
  HotelDetailsImages,
  HotelDetailsLocation,
  HotelDetailsPolicies,
  HotelDetailsPolicies2,
  HotelDetailsRoomTypes,
  HotelDetailsUpcomingBookings,
} from './assets/hotel-details';
import { useParams } from 'next/navigation';
import { useGetHotelByIdQuery } from '@/generated';
export const HotelDetails = () => {
  const { hotel } = useParams();
  const { data } = useGetHotelByIdQuery({ variables: { id: hotel as string } });
  const hotelDetails = data?.getHotelById;
  return (
    <DetailsContainer name={hotelDetails?.name || 'New Hotel'}>
      <DetailsLeft>
        <DetailsCard>
          <HotelDetailsUpcomingBookings />
        </DetailsCard>
        <DetailsCard>
          <HotelDetailsRoomTypes />
        </DetailsCard>
        <DetailsCard>
          <HotelDetailsGeneralInfo />
        </DetailsCard>
        <DetailsCard>
          <HotelDetailsAmenities />
        </DetailsCard>
        <DetailsCard>
          <HotelDetailsAboutThisProperty />
        </DetailsCard>
        <DetailsCard>
          <HotelDetailsPolicies />
        </DetailsCard>
        <DetailsCard>
          <HotelDetailsPolicies2 />
        </DetailsCard>
        <DetailsCard>
          <HotelDetailsFrequently />
        </DetailsCard>
      </DetailsLeft>
      <DetailsRight>
        <DetailsCard>
          <HotelDetailsLocation />
        </DetailsCard>
        <DetailsCard>
          <HotelDetailsImages images={hotelDetails?.images} />
        </DetailsCard>
      </DetailsRight>
    </DetailsContainer>
  );
};
