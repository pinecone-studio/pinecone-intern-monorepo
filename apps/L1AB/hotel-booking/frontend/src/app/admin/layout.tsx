import { Container } from "@/components/admin"

const AdminLayout = ({ children }: { children: React.ReactNode }) => {
    return (
        <Container>
            {children}
        </Container>
    )
}
export default AdminLayout