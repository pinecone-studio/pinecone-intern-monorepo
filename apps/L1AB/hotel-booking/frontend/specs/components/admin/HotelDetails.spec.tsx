import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { HotelDetails } from '@/components/admin';

describe("Admin Hotel Details", () => {
    it("should render the admin hotel details", () => {
        render(<HotelDetails />)
    })
})