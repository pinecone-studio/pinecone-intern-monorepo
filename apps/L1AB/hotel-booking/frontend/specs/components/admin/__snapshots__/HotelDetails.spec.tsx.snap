<div>
  <div
    data-testid="DetailsContainer"
  >
    <div
      data-testid="DetailsLeft"
    >
      <div
        data-testid="DetailsCard"
      >
        <div
          data-testid="HotelDetailsUpcomingBookings"
        >
          HotelDetailsUpcomingBookings
        </div>
      </div>
      <div
        data-testid="DetailsCard"
      >
        <div
          data-testid="HotelDetailsRoomTypes"
        >
          HotelDetailsRoomTypes
        </div>
      </div>
      <div
        data-testid="DetailsCard"
      >
        <div
          data-testid="HotelDetailsGeneralInfo"
        >
          HotelDetailsGeneralInfo
        </div>
      </div>
      <div
        data-testid="DetailsCard"
      >
        <div
          data-testid="HotelDetailsAmenities"
        >
          HotelDetailsAmenities
        </div>
      </div>
      <div
        data-testid="DetailsCard"
      >
        <div
          data-testid="HotelDetailsAboutThisProperty"
        >
          HotelDetailsAboutThisProperty
        </div>
      </div>
      <div
        data-testid="DetailsCard"
      >
        <div
          data-testid="HotelDetailsPolicies"
        >
          HotelDetailsPolicies
        </div>
      </div>
      <div
        data-testid="DetailsCard"
      >
        <div
          data-testid="HotelDetailsPolicies2"
        >
          HotelDetailsPolicies2
        </div>
      </div>
      <div
        data-testid="DetailsCard"
      >
        <div
          data-testid="HotelDetailsFrequently"
        >
          HotelDetailsFrequently
        </div>
      </div>
    </div>
    <div
      data-testid="DetailsRight"
    >
      <div
        data-testid="DetailsCard"
      >
        <div
          data-testid="HotelDetailsLocation"
        >
          HotelDetailsLocation
        </div>
      </div>
      <div
        data-testid="DetailsCard"
      >
        <div
          data-testid="HotelDetailsImages"
        >
          HotelDetailsImages
        </div>
      </div>
    </div>
  </div>
</div>
`;
